### PR TITLE
Enable Camera2D smoothing on limit change

### LINKED ISSUE #32596: Camera Limit smoothing does not work when new limit does not completely bound existing camera position limits
**Godot version:** 3.1/3.2a1

**OS/device including version:** Win10 x64

**Issue description:** 
When dynamically setting camera limits, if the limit is extended beyond current limits, the camera smoothing for the limit will slowly interpolate the camera to its new position. - This is the limit smoothing behavior working as expected.

When creating a new limit which does not fully contain the current camera view-port, the camera position snaps to its new position within the new limit, rather than smoothly interpolating to this new position.

**Steps to reproduce:**
Place camera at 0,0, enable limits at huge bounds (+/- 10000), enable limit smoothing.
Set script to change limit such that one of the X bounds bisects the default/unlimited camera position.
Camera will snap to new position in limit rather than smoothing to it.

**Minimal reproduction project:**

[Camera_Bug.zip](https://github.com/godotengine/godot/files/3694644/Camera_Bug.zip)

Camera starts limited, moving left will reach a border where the limits are massively extended (in effect - removed), moving back into the limited area will cause the camera to do the aforementioned snapping behavior due to limits being re-assigned.

### PR DESCRIPTION
**Pull request by ek68794998**
This change resolves #32596.

Currently, if the limiting bounds for a `Camera2D` are changed such that the camera is no longer in the bounds, the camera will jump (immediately) to the new bounds, even if `Smoothed` is checked for the `Camera2D` object. This change alters that behavior, allowing the camera to smoothly move to the new position when adjusting to new limits.

For scenarios where `Smoothed` is enabled but the "jump" behavior is desired, one should invoke `reset_smoothing()` after updating the limits.

*Bugsquad edit:* fixes #32596.

### CODE REVIEW

**Review by phocodes — APPROVED** [review]

**Comment by KoBeWi** [comment]
> For scenarios where Smoothed is enabled but the "jump" behavior is desired, one should invoke reset_smoothing() after updating the limits.

I think this should be mentioned in the docs, as someone might want the old behavior. Also `limit_smoothed` doesn't work without `smoothing_enabled`, so you could mention it as well (even if it's not directly related to the PR).

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Could you squash the two commits into one? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

**Comment by ekumlin** [comment]
@akien-mga I can give it a try, but this raises a question for me: Why does Godot not permit contributors to simply complete a PR using a squash merge?

image

**Comment by akien-mga** [comment]
That's our policy, we don't use squash in merge as it makes the history hard to follow. It's much nicer when the commits in `master` match the ones in the PR where they originated.

**Comment by ekumlin** [comment]
@akien-mga I see. Can't say I agree, but that's a separate discussion.

For now, please let me know if I did the rebase correctly. I'm not sure how to check if it's in a good state now.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by ekumlin** [comment]
Thanks @akien-mga!

Just to double-check &mdash; based on the PR workflow page, it looks like release managers and other contributors handle cherry-picking to stable branches. Is there any work I have to do here to help make that happen?

**Comment by akien-mga** [comment]
The PR has the `cherrypick:3.x` label, so it's a candidate for cherry picking and I'll do it eventually (for 3.4). If the code changed significantly and there are lots of conflicts to solve, you could make a backport PR for 3.x directly to save me some work, but in this case it should be fairly trivial to cherry pick for me.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.
